File: impl/rb/test/uxid_test.rb
require "test_helper"

class UXIDTest < Minitest::Test
  def test_that_it_has_a_version_number
    refute_nil ::UXID::VERSION
  end

  def test_generate_with_no_options
    uxid = UXID.generate

    assert_equal 26, uxid.length
  end

  def test_generate_with_size_and_prefix
    uxid = UXID.generate size: :xs, prefix: "foo"
    # foo_TTTTTTTTTT

    assert_equal 14, uxid.length
  end
end

File: impl/ts/test/uxid.test.ts
"use strict";
import UXID from "../src/uxid"

test("UXID", () => {
  expect(UXID).not.toBeNull()
})

test("UXID.generate()", () => {
  const uxid = UXID.generate()

  expect(uxid).not.toBeNull()
  expect(uxid.length).toBe(26)
})

test("UXID.generate(\"cus\")", () => {
  const uxid = UXID.generate("cus")

  expect(uxid).not.toBeNull()
  expect(uxid.length).toBe(30)
})

/*
test("UXID.generate(\"cus\", \"xs\")", () => {
  const uxid = UXID.generate("cus", "xs")
  console.log(uxid)

  expect(uxid).not.toBeNull()
  expect(uxid.length).toBe(14)
})

test("UXID.generate(\"cus\", \"xl\")", () => {
  const uxid = UXID.generate("cus", "xl")
  console.log(uxid)

  expect(uxid).not.toBeNull()
  expect(uxid.length).toBe(30)
})
*/
